### PR TITLE
More universal unit tests, ability to edit bootstrap logic

### PR DESCRIPTION
**Pull request by MarvinJanssen**
A quick PR that abstracts the bootstrap logic away from `generate-tests.ts`. It makes unit test generation more universal in case we want to use it in different locations in the future. The bootstrap function is now found in `tests/bootstrap.ts`. If the file exists and exports a `bootstrap` function, it will be called prior to executing a unit test. We use it to initialise the sBTC Mini protocol.

### CODE REVIEW

**Review by setzeus — APPROVED** [review]
Pulled & can confirm all unit tests still pass - LGTM.

**Review by jo-tm — APPROVED** [review]
Approving with a minor observation on the clarinet version, maybe change needed.

File: sbtc-mini/ext/generate-tests.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
 import { Clarinet, Contract, Account } from 'https://deno.land/x/clarinet@v1.5.4/index.ts';
```
[the remark this answers is not in the document]
Comment (MarvinJanssen) [reply sbtc-mini/ext/generate-tests.ts]: Doesn't make a difference in this case but doesn't hurt to update. Addressed in 325c12f on this PR.

File: sbtc-mini/ext/generate-tests.ts [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
 
+const sourcebootstrapFile = './tests/bootstrap.ts';
 const targetFolder = '.test';
 
 const warningText = `// Code generated using \`clarinet run ./scripts/generate-tests.ts\`
 // Manual edits will be lost.`;
 
+const defaultDeps = `import { Clarinet, Tx, Chain, Account, Block, types } from 'https://deno.land/x/clarinet@v1.5.4/index.ts';
```
[the remark this answers is not in the document]
Comment (MarvinJanssen) [reply sbtc-mini/ext/generate-tests.ts]: Resolved, see prior comment.

File: sbtc-mini/tests/bootstrap.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Tx, Chain, Account, types } from 'https://deno.land/x/clarinet@v1.5.4/index.ts';
```
[the remark this answers is not in the document]
Comment (MarvinJanssen) [reply sbtc-mini/tests/bootstrap.ts]: Resolved, see prior comment.

**Comment by jo-tm** [comment]
@sergey-shandar can we get back the `codecov` feature on the PRs that prints the table of coverage % ? :D

**Pull request merged**